### ISSUE TITLE
Authorize should return correct Grpc Status

### ISSUE THREAD

**Issue by buehler** [grpc/grpc-dotnet#264]
Hey there :-)

I'm currently playing around with the latest builds of the `Grpc.AspNetCore.Server`. What I've encountered when using authentication: the call does correctly use the authorization middleware but when the authorization fails (no token or invalid token) the call just "ends". 

It would be nice to return a RpcException with status `Unauthorized` so that clients can react to this.

```c#
throw new RpcException(new Status(StatusCode.Unauthorized, "Not authorized"));
```

What are your opinions on this?
Regards

**Comment by JamesNK** [comment]
I'm not sure whether it is appropriate to modify the response to return a different status code.

I tested what happens with a 401 response from `UseAuthorization` and a call doesn't end, but throws a cancel status with a message that a 401 status code was returned. I think that is fine.

If you really want to change `UseAuthorization` to return a gRPC status code then you could have your own `AuthenticationScheme` that sets a gRPC status code on the response when a call is challenged.

Alternatively you could add some middleware like this to change it:

```cs
app.UseRouting();

app.Use(async (context, next) =>
{
    await next();

    // If response is unauthorized and the endpoint is a gRPC method then
    // return grpc-status permission denied instead
    if (context.Response.StatusCode == StatusCodes.Status401Unauthorized &&
        context.GetEndpoint()?.Metadata.GetMetadata<GrpcMethodMetadata>() != null)
    {
        context.Response.StatusCode = StatusCodes.Status200OK;
        context.Response.AppendTrailer("grpc-status", StatusCode.PermissionDenied.ToString("D"));
    }
});

app.UseAuthentication();
app.UseAuthorization();

app.UseEndpoints(endpoints =>
{
    endpoints.MapGrpcService<MailerService>();
});
```

**Comment by buehler** [comment]
Hey @JamesNK 
Thank you for the answer. Yeah you are definitely right. I did not think about those consequences. It just seemed odd that this uses http statuscodes whereas the used technology is grpc which has its own statuscodes.

When I tested it, I used `BloomRPC` as a graphical client. And obviously this client has a bug so that unauthorized requests are running with cancelation.
I just tested the behaviour with `grpc_cli` and voila, the result is as expected.

So nevermind ;-)
